angular-bind-html-compile.js

```diff
@@ -6,15 +6,17 @@
     module.directive('bindHtmlCompile', ['$compile', function ($compile) {
         return {
             restrict: 'A',
-            link: function (scope, element, attrs) {
-                scope.$watch(function () {
+            link: function(scope, element, attrs) {
+                scope.$watch(function() {
                     return scope.$eval(attrs.bindHtmlCompile);
-                }, function (value) {
+                }, function(value) {
                     // Incase value is a TrustedValueHolderType, sometimes it
                     // needs to be explicitly called into a string in order to
                     // get the HTML string.
                     element.html(value && value.toString());
-                    $compile(element.contents())(scope);
+                    // If scope is provided use it, otherwise use parent scope
+                    var compileScope = scope.$eval(attrs.bindHtmlScope) || scope;
```

Comment:
I think we should probably wrap this in an if statement. We only want to attempt to eval `attrs.bindHtmlScope` if it exists?

e.g.
```
var compileScope = scope;
if (attrs.bindHtmlScope) {
    compileScope = scope.$eval(attrs.bindHtmlScope);
}
```